### ISSUE TITLE
Code highlighting fails if casting to a certain syntax of pointer

### ISSUE THREAD

**Issue by chuggafan** [microsoft/vscode-cpptools#3741]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 latest
- VS Code Version: 1.35
- C/C++ Extension Version: 0.23.1
- Other extensions you installed (and if the issue persists after disabling them): Git lens (issue persists)
- A clear and concise description of what the bug is.
Color highlighting fails for the rest of the line if you cast like this:

```cpp

void* func()
{
return nullptr;
}
enum var
{

};
void func2()
{
    enum var** thing = (enum var**)func();
}
```

For the rest of the line after the cast to enum var** the colorization does not work, this is the same if you use struct as the keyword instead of var. I think it has to do with expecting C++ style structure casting vs. C, although c++ can use both.

Here's a screenshot showing my example and its highlighting.
![syntax](https://user-images.githubusercontent.com/15003854/59116083-b6c38c80-8918-11e9-838f-f98f68a8e8b0.PNG)

**Comment by Colengms** [comment]
Hi @chuggafan,

Currently, colorization is provided by VS Code.  It uses TextMate grammar to colorize.  You might get improved results with <a HREF="https://marketplace.visualstudio.com/items?itemName=jeff-hykin.better-cpp-syntax">this extension</a>.

We are actually in the process of implementing colorization based on IntelliSense, which we hope to include in the next release of the C/C++ extension.

 - Colen

**Comment by jeff-hykin** [comment]
Temp fix was created just now on `v1.11.7` of the extension. There's likely a similar case with enums that will probably get fixed sometime this weekend.

**Comment by Colengms** [comment]
Closing as this is no longer repro'ing in VS Code.
